src/binary/ibc-setup/commands/init.ts

```diff
@@ -0,0 +1,57 @@
+import fs from 'fs';
+import path from 'path';
+
+import { Command } from 'commander';
+import yaml from 'js-yaml';
+
+import { GlobalOptions } from '../types';
+import { resolveOption } from '../utils/resolve-option';
+
+type Options = GlobalOptions & {
+  src: string;
+  dest: string;
+};
+
+type ProgramOptions = Partial<Options>;
+
+function getDefaultHome() {
+  if (!process.env.HOME) {
+    throw new Error('$HOME environment variable is not set.');
+  }
+
+  return `${process.env.HOME}/.ibc-setup`;
+}
+
+const REGISTRY_FILE = 'registry.yaml';
+
+export function init(_: unknown, program: Command) {
+  const programOptions: ProgramOptions = program.opts();
+
+  const options = {
+    src: resolveOption('src', programOptions.src, process.env.RELAYER_SRC),
+    dest: resolveOption('dest', programOptions.dest, process.env.RELAYER_DEST),
+    home: resolveOption('home', programOptions.home, getDefaultHome),
```

Comment:
I'm thinking of curry-ing the `resolveOption` function so we can inject logger or any other parameters in the future. Alternatively, it could accept options in an array.

```ts
const options = {
  src: resolveOption('src', logger)(programOptions.src, process.env.RELAYER_SRC)
  ...
}
```

or

```ts
const options = {
  src: resolveOption('src', 'logger', [programOptions.src, process.env.RELAYER_SRC])
  ...
}
```

Reply:
I think it is fine as it is now. I don't see the purpose of passing in a logger to such a small function with no I/O

Reply:
Alright, in case it's a quick refactor.